lib/ipfs/IpfsStorage.ts

```diff
@@ -23,13 +26,8 @@ export default class IpfsStorage {
       throw new SidetreeError(ErrorCode.IpfsStorageInstanceCanOnlyBeCreatedOnce,
         'IpfsStorage is a singleton thus cannot be created twice. Please use the getSingleton method to get the instance');
     }
-
-    const localRepoName = 'sidetree-ipfs';
-    const options = {
-      repo: repo !== undefined ? repo : localRepoName
-    };
-    const node = await IPFS.create(options);
-    IpfsStorage.ipfsStorageSingleton = new IpfsStorage(node);
+    const node = await IpfsStorage.getNode(repo);
+    IpfsStorage.ipfsStorageSingleton = new IpfsStorage(node, repo);
```

Comment:
Discussion: I find this initialization a bit confusing (the relationship between IpfsStorage, node, and repo) every time I come back to it after a month or so.

Do you think it would be cleaner if:
1. We only call `getNode()` internally (probably by monitor). and
1. Pass only repo into constructor.
1. Set `healthy` to false in  constructor. (and optionally rename `healthy` to perhaps `reinitializeRequired`)